apps/linode-marketplace-filecloud/roles/filecloud/tasks/main.yml

```diff
@@ -0,0 +1,96 @@
+# FileCloud installation
+- name: Adding Keys for MongoDB
+  get_url:
+      url: https://pgp.mongodb.com/server-6.0.asc
+      dest: /usr/share/keyrings/mongodb-server-6.0.asc
+      mode: '0644'
+      force: true 
+
+- name: adding mongodb repo
+  apt_repository:
+    repo: "deb [ arch=amd64,arm64 signed-by=/usr/share/keyrings/mongodb-server-6.0.asc ] https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/6.0 multiverse"
+    state: present
+
+- name: Adding Keys for FileCloud
+  get_url:
+      url: https://repo.filecloudlabs.com/static/pgp/filecloud.asc
+      dest: /usr/share/keyrings/filecloud.asc
+      mode: '0644'
+      force: true
+
+- name: adding filecloud repo
+  apt_repository:
+    repo: "deb [ arch=amd64 signed-by=/usr/share/keyrings/filecloud.asc ] https://repo.filecloudlabs.com/apt/ubuntu jammy/filecloud/23.232 main"
+    state: present
+
+- name: running apt-get update
+  apt:
+    update_cache: yes
+    upgrade: yes
+
+- name: Installing the Apache and MongoDB servers.
+  apt:
+    name:
+    - apache2
+    - mongodb-org
+    - pigz
+    state: present
+
+- name: Installing PHP 8.2 and dependencies
+  apt:
+    name: php8.2*
+    install_recommends: no
+    state: present
+
+- name: Installing FileCloud! :)
+  apt:
+    name: filecloud
+    state: present
+
+- name: grabbing private ip
+  shell: 
+    cmd: hostname -i
+  register: private_ip
+
+- name: logging into filecloud
+  shell:
+    chdir: /tmp/
+    cmd: curl -c login.txt --location --request POST 'http://{{ private_ip.stdout }}/admin/adminlogin?adminuser=admin&adminpassword=password'
+
+- name: getting value of tonidocloud-ah
+  shell:
+    chdir: /tmp/
+    cmd: awk 'NR > 3 {print $7}' login.txt | sed -n 2p
+  register: tonidocloud_ah
+
+- name: getting value of tonidocloud-as
+  shell:
+    chdir: /tmp/
+    cmd: awk 'NR > 3 {print $7}' login.txt | sed -n 3p
+  register: tonidocloud_as
+
+- name: getting value of x-xsrf-token
+  shell:
+    chdir: /tmp/
+    cmd: awk 'NR > 3 {print $7}' login.txt | sed -n 5p
+  register: x_xsrf_token
```

Comment:
Shell module can be avoided. Use slurp and then you can use regex_search() to parse what's needed.